src/python/pants/backend/python/rules/python_sources.py

```diff
@@ -134,7 +151,7 @@ async def prepare_unstripped_python_sources(
         )
     )
     source_root_paths = {source_root_obj.path for source_root_obj in source_root_objs}
-    return UnstrippedPythonSources(sources.snapshot, tuple(sorted(source_root_paths)))
+    return UnstrippedPythonSources(init_injected, tuple(sorted(source_root_paths)))
 
 
 def rules():
```

Comment:
Consider registering the init rules here in `python_sources.py`. It will avoid you having to update all those tests.

Reply:
Also, it would be good to check that this passes the new tests in https://github.com/pantsbuild/pants/pull/10519. I suspect that some of the backends would no longer load properly if `pants.backend.python` wasn't loaded too due to missing rules.

I'm hoping to land that tonight. If you land this first, I can fix the issues here.

Reply:
I still don't know what the policy is on where we register rules...

Reply:
I'll land this now if that's OK? After I make the fixes.

Reply:
The policy is to make sure that the new tests in https://github.com/pantsbuild/pants/pull/10519 pass.

Other than that, there is no policy. I personally recommend that util rules register their direct dependencies so that test sites don't have to register the entire universe. But we also have a very bad practice where, in production, we have far too many `RootRule`s just to make tests happy. We need to figure out a better solution still for how to make registration of rules ergonomic in tests while keeping the tests hermetic.